pkg/api/response/response.go

```diff
@@ -80,21 +80,11 @@ func (r *NormalResponse) ErrMessage() string {
 
 func (r *NormalResponse) WriteTo(ctx *contextmodel.ReqContext) {
 	if r.err != nil {
-		v := map[string]interface{}{}
-		traceID := tracing.TraceIDFromContext(ctx.Req.Context(), false)
-		if err := json.Unmarshal(r.body.Bytes(), &v); err == nil {
-			v["traceID"] = traceID
-			if b, err := json.Marshal(v); err == nil {
-				r.body = bytes.NewBuffer(b)
-			}
+		if errutil.HasUnifiedLogging(ctx.Req.Context()) {
+			ctx.Error = r.err
```

Comment:
I'm still fuzzy on parts of this PR. Is the idea that errors will only be captured by `ctx.Error` and that `ctx.Logger` will be used for other types of logs?

Reply:
Yeah, basically.

Not trying to fix anything beyond the somewhat standardized error logging at the moment.